lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -8,7 +8,9 @@ class NetHttpPersistent < NetHttp
       def net_http_connection(env)
         @cached_connection ||=
           if Net::HTTP::Persistent.instance_method(:initialize).parameters.first == [:key, :name]
-            Net::HTTP::Persistent.new(name: 'Faraday')
+            options = {}
+            options[:pool_size] = @connection_options[:pool_size] if @connection_options.key?(:pool_size)
+            Net::HTTP::Persistent.new(name: 'Faraday', **options)
```

Comment:
The above can be rewritten as:

```ruby
options = {name: 'Faraday'}.merge(@connection_options.slice(:name, :pool_size)
Net::HTTP::Persistent.new(options)
```

This way we:
* Simplify the filtering using `slice`
* Allow to override the connection name

`proxy` would also be a valid key, but I'd not slice it in as Faraday manages the proxy separately.

Reply:
I believe `slice` is only available in rails.

I've updated the the PR but only with a small improvement to fix the 1.9.3 issue. 
Out of time now but feel free to refactor if you have any other ideas.

Reply:
I tried it in the console to be sure and it worked, but I just found out that's because it was also added to Ruby 2.5 🤦‍♂️.
You're right, it's not available in Ruby 2.4 or before unless using activesupport.
The new implementation doesn't allow to customise `name`, but was never part of the issue you opened so I guess we can leave that out for now 👍